service/service.go

```diff
@@ -318,60 +277,43 @@ func (s *Service) GetUser(ctx context.Context, req *pb.UserRequest) (*pb.UserRes
 	logger.RequestService("GetUser")
 
 	if ok := serviceStateLocker.isStateAvailable(); !ok {
-		logger.Error("GetUser: ", errServiceUnavailable.Error())
-		return nil, status.Error(codes.Unavailable, errServiceUnavailable.Error())
+		logger.Error(getUserTag, errServiceUnavailable.Error())
+		return nil, statusServiceUnavailable
 	}
 
 	if req == nil {
-		return nil, status.Error(codes.InvalidArgument, errNilRequestUser.Error())
+		return nil, statusNilRequestUser
 	}
 
 	if err := refreshDBConnection(); err != nil {
-		return nil, status.Error(codes.Unknown, err.Error())
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
 	// get User Object
 	user := req.GetUser()
 	if user == nil {
 		logger.Error(errNilRequestUser.Error())
-		return nil, status.Error(codes.InvalidArgument, errNilRequestUser.Error())
-	}
-
-	// validate user id
-	if err := validateUUID(user.GetUuid()); err != nil {
-		logger.Error("GetUser validating uuid: ", err.Error())
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+		return nil, statusNilRequestUser
 	}
 
 	lock, _ := uuidMapLocker.LoadOrStore(user.GetUuid(), &sync.RWMutex{})
 	lock.(*sync.RWMutex).Lock()
```

Comment:
This should be locked for reading, because we are only getting or reading the user from DB